tc/sgx/trusted_worker_manager/enclave/CMakeWorkloads.txt

```diff
@@ -0,0 +1,94 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+##################################################################
+MACRO(CREATE_SUPPORTED_WORKLOADS_LIST)
+    SET(SUPPORTED_WORKLOADS_LIST "")
+    LIST(APPEND SUPPORTED_WORKLOADS_LIST "echo-result")
+    LIST(APPEND SUPPORTED_WORKLOADS_LIST "heart-disease-eval")
+    LIST(APPEND SUPPORTED_WORKLOADS_LIST "inside-out-eval")
+    LIST(APPEND SUPPORTED_WORKLOADS_LIST "simple-wallet")
+
+    # A new workload can be added by using below syntax
+    # LIST(SUPPORTED_WORKLOADS_LIST "<workload_id>")
+ENDMACRO()
+
+##################################################################
+MACRO(CREATE_SUPPORTED_WORKLOAD_LIBRARY_LIST)
+    SET(SUPPORTED_WORKLOAD_LIBRARY_LIST "")
+    LIST(APPEND SUPPORTED_WORKLOAD_LIBRARY_LIST "echo")
+    LIST(APPEND SUPPORTED_WORKLOAD_LIBRARY_LIST "heart_disease_eval")
+    LIST(APPEND SUPPORTED_WORKLOAD_LIBRARY_LIST "inside_out_eval")
+    LIST(APPEND SUPPORTED_WORKLOAD_LIBRARY_LIST "simple_wallet")
+
+    # A new workload library can be added by using below syntax
+    # Note: Order of workload id should match the order in
+    #       supported workload library list
+    # LIST(APPEND SUPPORTED_WORKLOAD_LIBRARY_LIST "<workload_lib_name>")
+ENDMACRO()
+
+##################################################################
+MACRO(CREATE_LINK_WORKLOAD_LIBRARY_LIST)
+    SET(WORKLOAD_LIST "$ENV{WORKLOADS}")
+    SET(WORKLOAD_LIB_LIST "")
+    # Create workload library list based on workloads passed
+    IF("${WORKLOAD_LIST}" STREQUAL "all" OR "${WORKLOAD_LIST} " STREQUAL " ")
+        SET(WORKLOAD_LIB_LIST ${SUPPORTED_WORKLOAD_LIBRARY_LIST})
+    ELSE()
+        FOREACH(workload ${WORKLOAD_LIST})
+            # Check if user provided workload is present in supported workload list
+            # If not found, return -1
+            LIST(FIND SUPPORTED_WORKLOADS_LIST ${workload} index)
+            IF(NOT ${index} EQUAL -1)
+                # Get workload library corresponding to workload identifier
+                LIST(GET SUPPORTED_WORKLOAD_LIBRARY_LIST ${index} workload_lib)
+                LIST(APPEND WORKLOAD_LIB_LIST ${workload_lib})
+            ELSE()
+                MESSAGE(FATAL_ERROR,
+                    "Unable to find workload library corresponding to parsed workload")
+            ENDIF()
+        ENDFOREACH(workload)
+    ENDIF()
+ENDMACRO()
+
+##################################################################
+MACRO(CONFIG_WORKLOAD_LINKER_FLAGS)
+    FOREACH(workload_lib ${WORKLOAD_LIB_LIST})
+        # workload folder name under $TCFHOME/example/apps/<workload_folder>
+        # must match the static library name configured inside workload CMakeLists.txt
```

Comment:
We should keep library name same as workload folder name not the other way around.
Let's rephrase above as workload static library name should match the workload folder name.
So instead of hard coding the library name in echo workload for example in https://github.com/hyperledger/avalon/blob/master/examples/apps/echo/workload/CMakeLists.txt#L25, we should use get the workload folder name using `get_filename_component()` as use this as static librray name in cmakelist.txt of workload.

Reply:
We also need to add a comment in workload CmakeList.txt that workload library name must match the workload folder name and this workload/library should be added in tc/sgx/trusted_worker_manager/enclave/CMakeWorkloads.txt.
And as suggested above can we use `get_filename_component()` to use workload directory name as workload static library name

Reply:
I will work on the documentation part. But using get_filename_component() to name library i will take it up in a separate pr, because this involves modifying all the workloads.

Reply:
Added comments in all workload CMakeLists.txt files and CMakeWorkloads.txt